### PR TITLE
Avoid dependency on local chromium/src checkout

### PR DESCRIPTION
**Pull request by mathiasbynens**
This patch avoids the dependency on a local chromium/src checkout.

Issue: #266

### CODE REVIEW

File: scripts/update-to-latest.sh [hunk 1]
```diff
@@ -20,33 +15,28 @@ git submodule update --init
...
+curl --silent "${browser_protocol_url}" | base64 --decode > pdl/browser_protocol.pdl
+curl --silent "${js_protocol_url}" | base64 --decode > pdl/js_protocol.pdl
 
 # extract cr revision number
-commit_pos_line=$(git log --date=iso --no-color --max-count=1 | gtac | grep -E -o "Cr-Commit-Position.*" | head -n1)
-commit_rev=$(echo "$commit_pos_line" | grep -E -o "\d+")
+chromium_git_log_url="https://chromium.googlesource.com/chromium/src.git/+log/refs/heads/main?format=JSON"
+commit_rev=$(curl --silent "${chromium_git_log_url}" | tail -n +2 | \
+  "$python_bin" -c 'import json,sys;print(json.load(sys.stdin)["log"][0]["message"])' | \
+  grep -E -o "^Cr-Commit-Position.*" | grep -E -o "\d+")
```
Comment (paulirish) [inline scripts/update-to-latest.sh:32]: i'm not 100% sure but i think i use `tac` to get the 'real' line when we have revert/reland commits which have 'cr-commit-position' multiple times in the message. i guess the `^` regex bit will do the same thing, assuming those nested messages always have the `>`. I'm fine with that assumption

File: scripts/update-to-latest.sh [hunk 2]
```diff
@@ -20,33 +15,28 @@ git submodule update --init
...
 
 # generate json from pdl
 convert_script="$protocol_repo_path/scripts/inspector_protocol/convert_protocol_to_json.py"
 "$python_bin" "$convert_script" --map_binary_to_string=true "$protocol_repo_path/pdl/browser_protocol.pdl" "$protocol_repo_path/json/browser_protocol.json"
 "$python_bin" "$convert_script" --map_binary_to_string=true "$protocol_repo_path/pdl/js_protocol.pdl" "$protocol_repo_path/json/js_protocol.json"
-# The conversion script leaves json files next to the pdl's. Because reasons.
-rm -f "$protocol_repo_path"/pdl/*.json
+# The conversion script leaves json files next to the PDLs. Because reasons.
+rm -f -- "$protocol_repo_path"/pdl/*.json
```
Comment (paulirish) [inline scripts/update-to-latest.sh:39]: ah. i like this. :)

**Review by paulirish — APPROVED** [review]
not dealing with a local chromium checkout is awesome.  going through these gerrit endpoints is a great idea.

File: scripts/update-to-latest.sh (hunk 1 above)
Comment (mathiasbynens) [reply scripts/update-to-latest.sh:32]: Yeah, that’s what I was thinking re: the `^`. I’m similarly unsure if this is actually enforced by Gerrit or whatever but it seemed reasonable to assume there’ll only be one `^Cr-Commit-Position` per commit message.

File: scripts/update-to-latest.sh [hunk 3]
```diff
@@ -20,33 +15,28 @@ git submodule update --init
...
-# copy the two protocol.pdl files over.
-cp "$js_protocol_path" "$protocol_repo_path/pdl"
-cp "$browser_protocol_path" "$protocol_repo_path/pdl"
+curl --silent "${browser_protocol_url}" | base64 --decode > pdl/browser_protocol.pdl
+curl --silent "${js_protocol_url}" | base64 --decode > pdl/js_protocol.pdl
 
 # extract cr revision number
-commit_pos_line=$(git log --date=iso --no-color --max-count=1 | gtac | grep -E -o "Cr-Commit-Position.*" | head -n1)
-commit_rev=$(echo "$commit_pos_line" | grep -E -o "\d+")
+chromium_git_log_url="https://chromium.googlesource.com/chromium/src.git/+log/refs/heads/main?format=JSON"
```
[the remark this answers is not in the document]
Comment (mathiasbynens) [reply scripts/update-to-latest.sh:29]: Ooh good catch! Will do.

**Pull request merged**